src/ERC20DripsHub.sol

```diff
@@ -127,37 +128,16 @@ contract ERC20DripsHub is ManagedDripsHub {
         return _setSplits(msg.sender, currReceivers, newReceivers);
     }
 
-    /// @notice Gets the the reserve where funds are stored.
-    function reserve() public view returns (IERC20Reserve) {
-        return IERC20Reserve(_reserveSlot().value);
-    }
-
-    /// @notice Set the new reserve address to store funds.
-    /// @param newReserve The new reserve.
-    function setReserve(IERC20Reserve newReserve) public onlyAdmin {
-        require(newReserve.erc20() == erc20, "Invalid reserve ERC-20 address");
-        IERC20Reserve oldReserve = reserve();
-        if (address(oldReserve) != address(0)) erc20.approve(address(oldReserve), 0);
-        _reserveSlot().value = address(newReserve);
-        erc20.approve(address(newReserve), type(uint256).max);
-        emit ReserveSet(oldReserve, newReserve);
-    }
-
-    function _reserveSlot() private pure returns (StorageSlot.AddressSlot storage) {
-        return StorageSlot.getAddressSlot(SLOT_RESERVE);
-    }
-
     function _transfer(address user, int128 amt) internal override {
-        IERC20Reserve erc20Reserve = reserve();
-        require(address(erc20Reserve) != address(0), "Reserve unset");
         if (amt > 0) {
             uint256 withdraw = uint128(amt);
-            erc20Reserve.withdraw(withdraw);
+            reserve.withdraw(withdraw);
             erc20.transfer(user, withdraw);
         } else if (amt < 0) {
             uint256 deposit = uint128(-amt);
             erc20.transferFrom(user, address(this), deposit);
-            erc20Reserve.deposit(deposit);
+            erc20.approve(address(reserve), deposit);
```

Comment:
It will be a bit more expensive to approve every time instead of the `approve uint.max`. 

However, I see the problem we would need some `register` function for first time usage of erc20. 

One idea could be have a blacklist of erc20 controlled by governance. The register function could check if an erc20 is the blacklist otherwise the approval happens here.

Reply:
Yeah, that's a solution to the unseen token approvals. We could check approved amount and set it to `uint.max` if it's different. That would probably remove a need for blacklists, IMO we don't need to protect from stealcoins with malicious implementations, if the users want to use it and they get rolled or the whole drips hub is stuck for that token, it's their responsibility. My plan here is to keep it as simple as possible because I want the funds to be transferred directly from the drivers to the reserve, so this code can be removed anyway.

It's not that much more expensive, in both cases the ERC20's `approved[dripsHub][reserve]` doesn't change. With `uint.max` that's true for all sane ERC-20 implementations where this value isn't updated (the standard doesn't enforce it). With `0 -> <deposit> -> 0` all storage costs are refunded because in the end nothing is changed. Sure, refunds aren't as convenient and reliable as not being charged in the first place.